plugins/etcd/src/main/java/org/radargun/service/EtcdService.java

```diff
@@ -0,0 +1,79 @@
+package org.radargun.service;
+
+import java.io.IOException;
+
+import org.radargun.Service;
+import org.radargun.config.Property;
+import org.radargun.traits.Lifecycle;
+import org.radargun.traits.ProvidesTrait;
+import org.radargun.utils.Utils;
+
+/**
+ *
+ * Usage instructions:
+ * You need to compile etcd on your target test system, and point to its directory with the distributionDir property.
+ * Alternatively you can zip the compiled distribution, point to the zip with the distributionZip property and RG will unzip it to distributionDir for you.
+ *
+ * Note that etcd saves some data in the distribution directory, and you need to delete it between runs. Using the distributionZip
+ * property is preferred, so the distribution is 'clean' with each run.
+ *
+ */
+@Service(doc = "Etcd")
+public class EtcdService implements Lifecycle {
+
+   private Process server;
+
+   @Property(doc = "Directory of the etcd distribution.", optional = false)
+   protected String distributionDir;
+
+   @Property(doc = "Distribution zip. If set, the zip will be extracted to the 'distributionDir' directory.")
+   protected String distributionZip;
+
+   @ProvidesTrait
+   public EtcdService getSelf() {
+      return this;
+   }
+
+   @Override
+   public void start() {
+
+      try {
```

Comment:
We should probably assert that distributionDir and distributionZip are not both set by the user. The behaviour would be unexpected.

Reply:
In this case, the `distributionZip` will be extracted into the `distributionDir`.
The `distributionDir` it will be `servers/server-0`, `servers/server-1`, `servers/server-2`. It will change in each slave. This idea comes from the `redis` plugin.

In the HotRod we have different attributes values but the same behavior.
`home` is `distributionDir`
`server-zip` is `distributionZip`

Reply:
OK. I get it now. The distributionDir is mandatory. The zip is optional. Cool.